### ISSUE TITLE
[css-images] symmetrical/mirrored css gradients

### ISSUE THREAD

**Issue by jsnkuhn** [w3c/csswg-drafts#7534]
There is usually an option for symmetrical gradients in images editors. We do not have this functionality in CSS gradients (https://www.w3.org/TR/css-images/).

SVG gradient `spreadMethod` already has a `reflect` option: 

> Reflect the gradient pattern start-to-end, end-to-start, start-to-end, etc. continuously until the target rectangle is filled.

https://svgwg.org/svg2-draft/pservers.html#LinearGradientElementSpreadMethodAttribute

Could allow devs the ability to write half the code and get the same result:

current:
```css
linear-gradient(transparent, red, red, transparent)
```

propsed:
```css
reflect/symmetrical/mirror?-linear-gradient(transparent, red 50%)
repeating-linear-gradient(transparent, red, reflect?)
```

Ana Tudor's example from twitter:

https://twitter.com/anatudor/status/1273657268106600449

https://codepen.io/thebabydino/pen/BajpyXV?editors=1100

**Comment by SebastianZ** [comment]
This definitely makes sense for gradients. Though I could imagine that this may also come in handy for other types of images to create some kind of pattern. So maybe this deserves a general solution for any kind of `<image>`?


Sebastian

**Comment by jsnkuhn** [comment]
This is something that has crossed my mind as well. I have a 2 player twitch layout with matching but reversed background images. I thought it would be handy if the even iteration of the `background-repeat` were rotated-y and then just `background-position` it into place like an auto generated sprite?
  
![image](https://user-images.githubusercontent.com/1286791/181129857-3ab1cd10-3ef1-436e-abc2-f8080f99f105.png)

seems like there is some cross over here with the `@image` proposal https://github.com/w3c/csswg-drafts/issues/6807

**Comment by SelenIT** [comment]
For vertical and horizontal gradients, this behavior can be currently emulated with radial gradient with extra large second radius ([forked example](https://codepen.io/SelenIT/pen/abYLLYg?editors=1100)). But I agree that a general non-hacky solution would be very useful!

**Comment by SebastianZ** [comment]
> seems like there is some cross over here with the `@image` proposal #6807

Yes, this use case already seems to be covered by the `transform` descriptor of `@image`, namely `transform: scaleX(-1);` or `transform: scaleY(-1);` depending on the mirroring direction.

We _could_ add this feature as a keyword to `background-repeat` and `mask-repeat` (and maybe other `*-repeat` properties in addition to that for the simple cases. The syntax might then look like this:

```css
[ repeat-x | repeat-y ] mirror? | [ [ repeat | space | round ] mirror? | no-repeat ]{1,2};
```

Sebastian

**Comment by jsnkuhn** [comment]
maybe more control with `mirror-x`, `mirror-y`, `mirror-both` options also? equating to rotating the even iteration on the x axis, y axis and both x and y respectively?

**Comment by jsnkuhn** [comment]
Occurs to me that many of the gradient patterns in use are about layering mirrored versions of a primary gradient. So a mirror syntax could probably simplify many of them. Examples from https://projects.verou.me/css3patterns/

checker board pattern (https://projects.verou.me/css3patterns/#checkerboard):

```css
background-image: linear-gradient(red 50%, #0000 50%);
background-size: 20px 20px;
background-repeat: repeat, mirror-x no-mirror;
```

diamond pattern (https://projects.verou.me/css3patterns/#diagonal-checkerboard):

```css
background-image: linear-gradient(to bottom right, gray 50%, silver 0);
background-size: 20px 20px;
background-repeat: repeat, mirror-y mirror-x;
```
